eth/api.go

```diff
@@ -409,11 +409,18 @@ type storageEntry struct {
 }
 
 // StorageRangeAt returns the storage at the given block height and transaction index.
-func (api *DebugAPI) StorageRangeAt(ctx context.Context, blockHash common.Hash, txIndex int, contractAddress common.Address, keyStart hexutil.Bytes, maxResult int) (StorageRangeResult, error) {
-	// Retrieve the block
-	block := api.eth.blockchain.GetBlockByHash(blockHash)
+func (api *DebugAPI) StorageRangeAt(ctx context.Context, blockNrOrHash rpc.BlockNumberOrHash, txIndex int, contractAddress common.Address, keyStart hexutil.Bytes, maxResult int) (StorageRangeResult, error) {
+	var block *types.Block
+	if number, ok := blockNrOrHash.Number(); ok {
```

Comment:
Or use `api.eth.APIBackend.BlockByNumberOrHash`. I'm not sure why we don't use backend so much in this file and directly use blockchain 🤔

Reply:
yeah that looks like a good alternative

Reply:
I moved `APIBackend.BlockByNumberOrHash/BlockByNumber` into the `Ethereum` object.

Reply:
Oops.  that was uneccesary.  `Ethereum` has a pointer to the `APIBackend`...

Reply:
Okay.  should be good now.